pkg/builder/kaniko/git.go

```diff
@@ -0,0 +1,47 @@
+package kaniko
+
+import (
+	"regexp"
+	"strings"
+)
+
+type GitContext struct {
+	Repo     string
+	Commit   string
+	Username string
+	Password string
+}
+
+func GetGitBuildContext(g *GitContext) (string, error) {
+	bCtx := ""
+
+	// cleaning the repo url
+	rgx, err := regexp.Compile(`^(https?|git|ssh|ftp)://`)
```

Comment:
I'd move the regexp string up to a const then make it easy to grab just the regexp and test/access